controllers/kserve_raw_inferenceservice_controller_test.go

```diff
@@ -0,0 +1,110 @@
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("The KServe Raw reconciler", func() {
+	var testNs string
+	createServingRuntime := func(namespace, path string) *kservev1alpha1.ServingRuntime {
+		servingRuntime := &kservev1alpha1.ServingRuntime{}
+		err := convertToStructuredResource(path, servingRuntime)
+		Expect(err).NotTo(HaveOccurred())
+		servingRuntime.SetNamespace(namespace)
+		if err := cli.Create(ctx, servingRuntime); err != nil && !apierrs.IsAlreadyExists(err) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+		return servingRuntime
+	}
+
+	createInferenceService := func(namespace, name string, path string) *kservev1beta1.InferenceService {
+		inferenceService := &kservev1beta1.InferenceService{}
+		err := convertToStructuredResource(path, inferenceService)
+		Expect(err).NotTo(HaveOccurred())
+		inferenceService.SetNamespace(namespace)
+		if len(name) != 0 {
+			inferenceService.Name = name
+		}
+		inferenceService.Annotations = map[string]string{}
+		inferenceService.Annotations["serving.kserve.io/deploymentMode"] = "RawDeployment"
+		return inferenceService
+	}
+
+	BeforeEach(func() {
+		testNs = Namespaces.Create(cli).Name
+
+		inferenceServiceConfig := &corev1.ConfigMap{}
+		Expect(convertToStructuredResource(InferenceServiceConfigPath1, inferenceServiceConfig)).To(Succeed())
+		if err := cli.Create(ctx, inferenceServiceConfig); err != nil && !apierrs.IsAlreadyExists(err) {
+			Fail(err.Error())
+		}
+
+	})
+
+	When("deploying a Kserve RawDeployment model with route disabled", func() {
+		It("it should create a clusterrolebinding for auth but not create a route", func() {
+			_ = createServingRuntime(testNs, KserveServingRuntimePath1)
+			inferenceService := createInferenceService(testNs, KserveOvmsInferenceServiceName, KserveInferenceServicePath1)
+			if err := cli.Create(ctx, inferenceService); err != nil && !apierrs.IsAlreadyExists(err) {
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			crb := &rbacv1.ClusterRoleBinding{}
+			Eventually(func() error {
+				key := types.NamespacedName{Name: inferenceService.Namespace + "-" + constants.KserveServiceAccountName + "-auth-delegator",
+					Namespace: inferenceService.Namespace}
+				return cli.Get(ctx, key, crb)
+			}, timeout, interval).ShouldNot(HaveOccurred())
+
+			route := &routev1.Route{}
+			Eventually(func() error {
+				key := types.NamespacedName{Name: inferenceService.Name, Namespace: inferenceService.Namespace}
+				return cli.Get(ctx, key, route)
+			}, timeout, interval).Should(HaveOccurred())
+		})
+	})
+	When("namespace no longer has any RawDeployment models", func() {
+		It("should delete the clusterrolebinding", func() {
+			_ = createServingRuntime(testNs, KserveServingRuntimePath1)
+			inferenceService := createInferenceService(testNs, KserveOvmsInferenceServiceName, KserveInferenceServicePath1)
+			if err := cli.Create(ctx, inferenceService); err != nil && !apierrs.IsAlreadyExists(err) {
```

Comment:
for these ifs I would suggest removing it, and just check with Expect, it is valid for the other test cases.
If issues are happening with duplicated resources, use a different name, it might be the cause of the timeout happening in this test.